src/metabase/lib/metadata.cljc

```diff
@@ -155,7 +163,9 @@
 
 (def MetadataProvider
   "Schema for something that satisfies the [[lib.metadata.protocols/MetadataProvider]] protocol."
-  [:fn lib.metadata.protocols/metadata-provider?])
+  [:fn
+   {:error/message "Valid MetadataProvider"}
```

Comment:
If you don't specify an `:error/message` or `:error/fn`, a `:fn` schema just gives some sort of unhelpful generic `Unknown error` message